### ISSUE TITLE
Separation of tasks and capabilites

### ISSUE THREAD

**Issue by Gautam-Rajeev** [nandakishore2404/oan-pest-detection-benchmark#1]
@nandakishore2404 

I think we should restructure the documentation of the repo so that we clearly specify what we are measuring 

For example I can see that there are multiple capabilites that we expect that we are conflating : 

ID | Capability | Input | Expected output
-- | -- | -- | --
A | Pest object detection | Image containing visible pests | Pest bounding boxes
B | Pest classification | Visible pest | Pest type/species
C | Disease/lesion detection | Leaf/plant image | Bounding box or segmentation of affected region
D | Disease classification | Leaf/plant showing symptoms | Disease type
E | Pest-damage diagnosis | Damaged plant where pest may not be visible | Likely causal pest
F | General crop diagnosis | Arbitrary farmer crop image | Pest, disease, healthy, deficiency, unknown, etc.

Metrics also can be task-specific if we cannot come up with a single metric that makes sense across all 

- A/C: mAP50:95, recall, per-class AP
- B/D: accuracy, macro-F1, per-class recall

A–D are conventional computer-vision tasks.

**E is most likley for our use case where farmers often photograph the affected leaf or plant rather than the insect itself.

---

Have just mapped the models to the tasks that we saw: 

| Model family | Pest detection (A) | Pest classification (B) | Lesion detection (C) | Disease classification (D) | Damage → pest (E) | General diagnosis (F) |
|---|---|---|---|---|---|---|
| YOLO / RF-DETR / RT-DETR | Yes | Yes, if trained | Yes, if trained | Limited | No | No |
| CNN / ViT classifier | No | Yes | No | Yes | Yes, if trained | Limited |
| Grounding DINO / open-vocabulary detector | Yes | Yes | Yes | Limited | No | Limited |
| Segmentation models | Yes | No | Yes | No | No | No |
| Vision-language models | Yes, with limitations | Yes | Possible | Yes | Yes | Yes |

This means we should not treat pest diagnosis as a single model-selection problem.

Object detectors are appropriate when the pest itself is visible. Classification models can work well when the task and taxonomy are known in advance. VLMs may already have enough knowledge to perform these tasks end to end 

---

Similarly, mapping our dataset to it: 

| Dataset / benchmark | Capabilities measured | Size used in repo | Data | Coverage | Image conditions | Appropriate models |
|---|---|---:|---|---|---|---|
| 28-class Agricultural Pests | A + B | 28 pest classes | Images containing visible insects/pests | Agrotis, Helicoverpa, Chilo, Nilaparvata, Spodoptera, etc. | Mixed / natural backgrounds | YOLO, RF-DETR, RT-DETR, Grounding DINO |
| Generic 12-class insect dataset | A + B | 546 test, 1,095 validation | Images containing visible insects | 12 insect classes | Mixed | Object detectors |
| 5-class foliar dataset | D | 83 test images | Leaves showing disease symptoms | Five disease/healthy classes | Relatively clean | CNN, ViT, VLM |
| African Beans Field | D | Field validation set | Real bean foliage | Angular leaf spot, bean rust, healthy | Real field images | CNN, ViT, VLM |
| Real field foliar evaluation | D | 261 | Real field foliage | Foliar disease | Noisy / uncontrolled | CNN, ViT, VLM |
| Fine-tuned field split | D | 42 | Real field foliage | Foliar disease | Noisy / uncontrolled | CNN, ViT, VLM |
| `benchmark_test_15` | A + B | 15 | Wild photographs containing pests | 28-class pest taxonomy | Wild / noisy | Object detectors |
| `dataset.csv` diagnostic set | A/B + D + healthy | 5 | Mixed pest, disease and healthy images | Fall Armyworm, three diseases, healthy | Mixed | Combined pipeline / VLM |
| Pest-damage-only benchmark | E | Missing | Damaged crop with no visible pest | Missing | Should be real field data | Classifier, VLM |
| General farmer-photo benchmark | F | Missing | Arbitrary farmer-uploaded images | Missing | Real user distribution | VLM / combined system |

The current benchmark therefore has coverage of:

- visible pest detection and classification;
- leaf disease classification;
- some field/domain-shift evaluation.

It has less coverage for:

- diagnosing the causal pest from damage when the pest is not visible;
- arbitrary farmer-submitted images;
- non-pest/non-disease cases such as nutrient deficiency;
- poor-quality or ambiguous images;

---
Farmers usually upload images like this 

```text
        Farmer image
             |
             v
       What is wrong?
             |
     -----------------
     |       |       |
    Pest   Disease  Other
     |       |       |
   Which?  Which?  Healthy /
                  deficiency /
                    unknown
```

This corresponds to capability F and is not currently represented by a meaningful benchmark, and we need to evaluate what combination of models give us this capability and what is the accuracy of this overall system 

---

The current results show a large difference between controlled evaluation and real-field performance:

| Evaluation | Reported result |
|---|---:|
| Foliar classifier on relatively clean/lab data | 87.95% |
| Foliar classifier on real field data | 51.7% |
| Fine-tuned field evaluation | 59.52% |
| Generic insect detector | 16–18% |
| 28-class pest detector on wild images | 0/15 |

So the current models are doing poorly on actual farmer images and will do poorly in production use cases. 

---
We should maybe have our own custom dataset that has: 

- crop damage where the causal pest is known;
- no visible pest in the image;
- real field conditions;
- multiple crops and pest species;
- different damage severity and plant stages;
- ambiguous cases where multiple pests could plausibly cause similar damage.
- visible pests;
- pest damage without visible pests;
- diseases;
- nutrient deficiencies;
- healthy plants;
- beneficial/non-pest insects;
- multiple simultaneous problems;
- blurry or badly framed photographs;
- unrelated images;
- cases where diagnosis is impossible from the provided image.

This should become the main end-to-end benchmark for production model selection.

---

I would also suggesting testing some of the bigger VLMs for this

Rather than evaluating many similar architectures, we should choose representative models from different approaches.

| Approach | Candidate | Primary capabilities |
|---|---|---|
| Open VLM | Qwen-VL family | B, D, E, F |
| Open VLM | Gemma Vision / InternVL family | B, D, E, F |
| Frontier VLM | GPT/Gemini-class vision model | B, D, E, F |

We can meaningfully compare specialized detectors/classifiers against modern VLMs and determine which approach—or combination of approaches—should be used in production.
